src/runtime/ScriptFunctionObject.cpp

```diff
@@ -77,6 +78,47 @@ NEVER_INLINE void ScriptFunctionObject::generateByteCodeBlock(ExecutionState& st
     auto& currentCodeSizeTotal = state.context()->vmInstance()->compiledByteCodeSize();
     ASSERT(currentCodeSizeTotal < std::numeric_limits<size_t>::max());
     currentCodeSizeTotal += interpretedCodeBlock()->byteCodeBlock()->memoryAllocatedSize();
+
+
+    if (hasTag(g_scriptFunctionObjectTag)) {
+        auto cb = m_codeBlock->asInterpretedCodeBlock();
+        auto byteCb = cb->byteCodeBlock();
+        size_t registerSize = byteCb->m_requiredRegisterFileSizeInValueSize;
+        size_t identifierOnStackCount = cb->identifierOnStackCount();
+        size_t stackStorageSize = cb->totalStackAllocatedVariableSize();
+        size_t literalStorageSize = byteCb->m_numeralLiteralData.size();
+        size_t registerFileSize = registerSize + stackStorageSize + literalStorageSize;
+
+        bool isStrict = cb->isStrict();
+        bool shouldClearStack = byteCb->m_shouldClearStack;
+
+#define DEFINE_SCRIPTSIMPLEFUNCTION_BRANCH(opt1, opt2)                   \
+    if (registerFileSize <= 4) {                                         \
+        writeTag(ScriptSimpleFunctionObject<opt1, opt2, 4>().getTag());  \
```

Comment:
I'm just wondering that it is really safe to rewrite the vtable address in the middle.
I'm not sure but would it incur any undesired side effect?

Reply:
IMO If sizeof ScriptSimpleFunctionObject and ScriptFunctionObject are same, rewrite is safe.
The only different content of vtable is ::call method address

Reply:
Ok I see!